parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -78,8 +87,8 @@ public void mergeStatisticsMinMax(Statistics stats) {
    */
   @Override
   public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes) {
-    max = Binary.fromReusedByteArray(maxBytes);
-    min = Binary.fromReusedByteArray(minBytes);
+    max = truncator.truncateMax(Binary.fromReusedByteArray(maxBytes), truncateLength);
```

Comment:
This method is invoked when we read the values from a file. It does not make sense truncating the values in the read path.

Reply:
Agree! Removed it.